src/foundry/common/types.d.mts

```diff
@@ -155,40 +162,52 @@ declare global {
 
     // TODO: This can take data models and data fields now
     /** The JS Type that the Setting is storing */
-    type?: T extends string
-      ? typeof String
-      : T extends number
-        ? typeof Number
-        : T extends boolean
-          ? typeof Boolean
-          : T extends Array<any>
-            ? typeof Array
-            : ConstructorOf<T>;
+    type?:
+      | (T extends string
+          ? typeof String
+          : T extends number
+            ? typeof Number
+            : T extends boolean
+              ? typeof Boolean
+              : T extends Array<any>
+                ? typeof Array
+                : ConstructorOf<T>)
+      | undefined;
 
     /** For string Types, defines the allowable values */
     choices?: (T extends number | string ? Record<T, string> : never) | undefined;
 
     /** For numeric Types, defines the allowable range */
-    range?: T extends number
-      ? {
-          max: number;
-          min: number;
-          step: number;
-        }
-      : never;
+    range?:
+      | (T extends number
+          ? {
+              max: number;
+              min: number;
+              step: number;
+            }
+          : never)
+      | undefined;
 
     /** The default value */
     default: T;
 
     /** Whether setting requires Foundry to be reloaded on change  */
-    requiresReload?: boolean;
+    requiresReload?: boolean | undefined;
 
     /** Executes when the value of this Setting changes */
-    onChange?: (value: T) => void;
+    onChange?: ((value: T) => void) | undefined;
 
     filePicker?: (T extends string ? true | "audio" | "image" | "video" | "imagevideo" | "folder" : never) | undefined;
```

Comment:
pretty sure `input` is replacing `filePicker` here

Reply:
That makes sense. I noticed I couldn't find `filePicker` in v11 either so I wasn't sure what to do with this one. Should I go ahead and remove this then?

Reply:
Yeah definitely remove